docs/reference/configuration/configuration.md

```diff
@@ -1,48 +1,276 @@
-            "thread-stack-size": {
-                "value": 2000
-            }
+| Section | Required | Meaning |
+| ------- | -------- | ------- |
+| `name`  | Yes      | Name of the library. Must be unique. May not be `app` or `target`.|
+| `macros` | No | List of macros to define on the command-line. |
+| `config` | No | Configuration parameters defined for use in this library. |
+| `target_overrides` | No | Overrides for target configuration parameters and configuration parameters of the current library. |
+
+The following is an example library, `mylib`.
+
+```JSON
+{
+    "name": "mylib",
+    "config": {
+        "buffer_size": 1024,
+        "timer_period": {
+            "help": "The timer period (in us)",
+            "macro_name": "INTERNAL_GPTMR_PERIOD",
+            "required": true
+        },
+        "queue_size": {
+            "help": "Size of event queue (entries)",
+            "value": 10
+        }
+    },
+    "macros": ["MYMOD_MACRO1", "MYMOD_MACRO2=\"TEST\""],
+    "target_overrides": {
+        "K64F": {
+             "timer_period": 100,
+             "queue_size": 40
+        },
+        "NXP": {
+             "queue_size": 20,
+             "buffer_size": 128,
+             "target.features_add": ["IPV4"]
         }
     }
-    ```
+}
+```
+
+In this JSON file:
+
+- `name` is the name of the library. **This is a required field.**
+- `config` defines the configuration parameters of the library, as the section about [defining configuration parameters](#defining-configuration-parameters) explains.
+- `macros` is a list of extra macros that are defined when compiling a application that includes this library. `macros` defines the provided list of C preprocessor macros.
+- `target_overrides` is a dictionary with target-specific values for the configuration parameters.
 
-2. Construct the thread like this in your application:
+All configuration parameters defined in `mylib` have a `mylib.` prefix. In `mbed_app.json`, `buffer_size` is accessible using the name `mylib.buffer_size`.
 
-    `Thread thread(osPriorityNormal, 2000);`
+Use `target_overrides` to override the values of the parameters, depending on the current compilation target. The configuration system matches keys in `target_overrides` against target labels. (You can find a description of Mbed targets in our documentation about [adding and configuring targets](/docs/development/tools/adding-and-configuring-targets.html).) If a key inside `target_overrides` matches one of the target labels, the parameter values change according to the value of the key. 
 
-By default, mbed OS dynamically allocates the memory for the thread's stack from the global heap, though you can change it in the [Thread constructor](/docs/development/reference/thread.html). Here is an example application that uses memory allocated from the main thread's stack for the newly spawned thread stack:
+It is an error for `mbed_lib.json` to override configuration parameters that were not defined.
```

Comment:
Tense looks a bit funny at the end? Thinking it should be:

"It is an error for `mbed_lib.json` to override configuration parameters that **are** not defined."